File: better_player-0.0.84/ios/Classes/BetterPlayerHlsTokenStore.m
#import "BetterPlayerHlsTokenStore.h"

@implementation BetterPlayerHlsTokenStore

static NSString *_hlsToken = nil;
static NSString *_hlsExp = nil;

+ (void)updateToken:(NSString *)token exp:(NSString *)exp {
    @synchronized(self) {
        _hlsToken = token;
        _hlsExp = exp;
    }
}

+ (NSString *)token {
    @synchronized(self) {
        return _hlsToken;
    }
}

+ (NSString *)exp {
    @synchronized(self) {
        return _hlsExp;
    }
}

+ (BOOL)hasToken {
    @synchronized(self) {
        return _hlsToken != nil && _hlsToken.length > 0 && _hlsExp != nil && _hlsExp.length > 0;
    }
}

@end

File: better_player_plus-1.1.5/ios/Classes/BetterPlayerHlsTokenStore.swift
import Foundation

enum BetterPlayerHlsTokenStore {
    private static let lock = NSLock()
    private static var tokenValue: String?
    private static var expValue: String?

    static func update(token: String?, exp: String?) {
        lock.lock()
        tokenValue = token
        expValue = exp
        lock.unlock()
    }

    static func token() -> String? {
        lock.lock()
        let value = tokenValue
        lock.unlock()
        return value
    }

    static func exp() -> String? {
        lock.lock()
        let value = expValue
        lock.unlock()
        return value
    }

    static func hasToken() -> Bool {
        guard let token = token(), !token.isEmpty,
              let exp = exp(), !exp.isEmpty else {
            return false
        }
        return true
    }
}
